### ISSUE TITLE
freerdp client mouse pointer issue (not tracking/syncing) ?  

### ISSUE THREAD

**Issue by Teraya** [FreeRDP/FreeRDP#9608]
I can see the pointer move and I can see mouse "clicks" are happening, they just don't seem to hit where the pointer is actually pointing.  

Tried with latest git clone repo code and compiling.  
Also tried installing flatpak 3.0.0-rc0 (flathub-beta).  
Also tried from two different versions of Fedora on two very different computers.  
Also tried changing wayland to X11 again for gdm.  
Also tried different combinations of mouse grab & relative mouse movement.  
Using relative mouse makes it "less bad", but still not usable.  
Also tried xfreerdp & sdl-freerdp client.  (If I try the sdl-freerdp client instead, it does the same thing bug the rate of drift is not quite as extreme).  
Also tried compiling with cairo on & swscale off, and again with cairo off & swscale on.  

I am able to see a mouse pointer and move it around properly, but if I right-click, the resulting context menu appears in a very different location.  

If I pull the mouse to the top-left of the screen for a bit, the visible pointer and actual pointer seem to align, but as I slowly move the pointer I can tell (by observing the focus changes on various desktop icons) that while I see the visible pointer move a fraction of an inch, the actual pointer moves several inches.  

It is as if the visible mouse cursor moves at a slower pace than the "real" pointer.  

This is with a command like...  xfreerdp xxx.rdp /d:'' /u:'x@x.x' /mouse:relative:on,grab:on +mouse-relative /w:1600 /h:900

I also tried some various sizes and size related options.  
  /w:1600 /h:900 /size:1600x900 /smart-sizing:1600x900
  /w:1368 /h:768 /size:1368x768 /smart-sizing:1368x768
  /w:1280 /h:800 /size:1280x800 /smart-sizing:1280x800
  /w:1024 /h:768 /size:1024x768 /smart-sizing:1024x768

This is while connected to a Microsoft Azure Virtual Desktop (AVD).  

It looks like others have encountered the same issue, here's one reference...  
"              _@akallabeth Thanks. I'm able to connect and get onto my desktop now. Next issue is the mouse. With /mouse:grab:on I'm stuck in the bottom right corner. With /mouse:relative:on I can move around but the mouse pointer and the location of the click are not in sync. I'll look for an existing bug or open a new one on that._
"  Originally posted by @imbezol in https://github.com/FreeRDP/FreeRDP/issues/6529#issuecomment-1803935166

$ /opt/freerdp/bin/xfreerdp /buildconfig
This is FreeRDP version 3.0.0-dev6 (dd56308)
Build configuration: BUILD_TESTING=OFF WINPR_HAVE_AIO_H=1 WINPR_HAVE_EXECINFO_BACKTRACE=1 WINPR_HAVE_EXECINFO_BACKTRACE_SYMBOLS=1 WINPR_HAVE_EXECINFO_BACKTRACE_SYMBOLS_FD=1 WINPR_HAVE_EXECINFO_HEADER=1 WINPR_HAVE_FCNTL_H=1 WINPR_HAVE_GETLOGIN_R=1 WINPR_HAVE_GETPWUID_R=1 WINPR_HAVE_INTTYPES_H=1 WINPR_HAVE_POLL_H=1 WINPR_HAVE_PTHREAD_MUTEX_TIMEDLOCK_LIB=1 WINPR_HAVE_PTHREAD_MUTEX_TIMEDLOCK_LIBS= WINPR_HAVE_PTHREAD_MUTEX_TIMEDLOCK_SYMBOL=1 WINPR_HAVE_STDBOOL_H=1 WINPR_HAVE_STDINT_H=1 WINPR_HAVE_STRNDUP=1 WINPR_HAVE_SYSLOG_H=1 WINPR_HAVE_SYS_EVENTFD_H=1 WINPR_HAVE_SYS_FILIO_H= WINPR_HAVE_SYS_SELECT_H=1 WINPR_HAVE_SYS_SOCKIO_H= WINPR_HAVE_SYS_TIMERFD_H=1 WINPR_HAVE_TM_GMTOFF=1 WINPR_HAVE_UNISTD_H=1 WINPR_HAVE_UNWIND_H=1 WITH_AAD=ON WITH_ADD_PLUGIN_TO_RPATH=OFF WITH_ALSA=ON WITH_CAIRO=ON WITH_CCACHE=ON WITH_CHANNELS=ON WITH_CLANG_FORMAT=ON WITH_CLIENT=ON WITH_CLIENT_AVAILABLE=1 WITH_CLIENT_CHANNELS=ON WITH_CLIENT_CHANNELS_AVAILABLE=1 WITH_CLIENT_COMMON=ON WITH_CLIENT_INTERFACE=OFF WITH_CLIENT_SDL=ON WITH_CLIENT_SDL_AVAILABLE=1 WITH_CUPS=ON WITH_DEBUG_ALL=OFF WITH_DEBUG_CAPABILITIES=OFF WITH_DEBUG_CERTIFICATE=OFF WITH_DEBUG_CHANNELS=OFF WITH_DEBUG_CLIPRDR=OFF WITH_DEBUG_CODECS=OFF WITH_DEBUG_DVC=OFF WITH_DEBUG_EVENTS=OFF WITH_DEBUG_KBD=OFF WITH_DEBUG_LICENSE=OFF WITH_DEBUG_MUTEX=OFF WITH_DEBUG_NEGO=OFF WITH_DEBUG_NLA=OFF WITH_DEBUG_NTLM=OFF WITH_DEBUG_RAIL=OFF WITH_DEBUG_RDP=OFF WITH_DEBUG_RDPDR=OFF WITH_DEBUG_RDPEI=OFF WITH_DEBUG_RDPGFX=OFF WITH_DEBUG_REDIR=OFF WITH_DEBUG_RFX=OFF WITH_DEBUG_RINGBUFFER=OFF WITH_DEBUG_SCARD=OFF WITH_DEBUG_SCHANNEL=OFF WITH_DEBUG_SDL_EVENTS=OFF WITH_DEBUG_SDL_KBD_EVENTS=OFF WITH_DEBUG_SND=OFF WITH_DEBUG_SVC=OFF WITH_DEBUG_SYMBOLS=OFF WITH_DEBUG_THREADS=OFF WITH_DEBUG_TIMEZONE=OFF WITH_DEBUG_TRANSPORT=OFF WITH_DEBUG_TSG=OFF WITH_DEBUG_TSMF=OFF WITH_DEBUG_TSMF_AVAILABLE=0 WITH_DEBUG_URBDRC=OFF WITH_DEBUG_WND=OFF WITH_DEBUG_X11=OFF WITH_DEBUG_X11_LOCAL_MOVESIZE=OFF WITH_DEBUG_XV=OFF WITH_DSP_EXPERIMENTAL=OFF WITH_DSP_FFMPEG=OFF WITH_DSP_FFMPEG_AVAILABLE=0 WITH_EVENTFD_READ_WRITE=1 WITH_FAAC=ON WITH_FAAD2=ON WITH_FFMPEG=OFF WITH_FREERDP_DEPRECATED=OFF WITH_FREERDP_DEPRECATED_COMMANDLINE=OFF WITH_FUSE=ON WITH_GFX_H264=ON WITH_GPROF=OFF WITH_GSM=ON WITH_INTERNAL_MD4=OFF WITH_INTERNAL_MD5=OFF WITH_INTERNAL_RC4=OFF WITH_IPP=OFF WITH_JPEG=ON WITH_KRB5=ON WITH_KRB5_NO_NTLM_FALLBACK=OFF WITH_LAME=OFF WITH_LIBRARY_VERSIONING=ON WITH_LODEPNG=OFF WITH_MACAUDIO=OFF WITH_MACAUDIO_AVAILABLE=0 WITH_MANPAGES=ON WITH_MBEDTLS=OFF WITH_NATIVE_SSPI=ON WITH_OPENCL=OFF WITH_OPENH264=TRUE WITH_OPENH264=TRUE WITH_OPENH264_LOADING=OFF WITH_OPENSC_PKCS11_LINKED=OFF WITH_OPENSSL=ON WITH_OPUS=ON WITH_OSS=ON WITH_PCSC=ON WITH_PKCS11=ON WITH_PLATFORM_SERVER=OFF WITH_POLL=ON WITH_PROFILER=OFF WITH_PULSE=ON WITH_SAMPLE=OFF WITH_SANITIZE_ADDRESS=OFF WITH_SANITIZE_ADDRESS_AVAILABLE=1 WITH_SANITIZE_MEMORY=OFF WITH_SANITIZE_MEMORY_AVAILABLE=1 WITH_SANITIZE_THREAD=OFF WITH_SANITIZE_THREAD_AVAILABLE=1 WITH_SERVER=OFF WITH_SERVER_INTERFACE=ON WITH_SMARTCARD_EMULATE=ON WITH_SMARTCARD_INSPECT=OFF WITH_SMARTCARD_PCSC=ON WITH_SOXR=ON WITH_SSE2=ON WITH_SWSCALE=OFF WITH_SYSTEMD=ON WITH_THIRD_PARTY=OFF WITH_UNICODE_BUILTIN=OFF WITH_URIPARSER=ON WITH_VAAPI=OFF WITH_VAAPI_AVAILABLE=0 WITH_VALGRIND_MEMCHECK=OFF WITH_VALGRIND_MEMCHECK_AVAILABLE=1 WITH_VERBOSE_WINPR_ASSERT=ON WITH_VIDEO_FFMPEG=OFF WITH_VIDEO_FFMPEG_AVAILABLE=0 WITH_WAYLAND=ON WITH_WEBVIEW=OFF WITH_WEBVIEW_QT=OFF WITH_WINPR_DEPRECATED=OFF WITH_WINPR_TOOLS=ON WITH_WIN_CONSOLE=ON WITH_X11=ON WITH_XCURSOR=ON WITH_XEXT=ON WITH_XFIXES=ON WITH_XI=ON WITH_XINERAMA=ON WITH_XRANDR=ON WITH_XRENDER=ON WITH_XV=ON
Build type:          Release
CFLAGS:               -Wall -Wpedantic -Wno-padded -Wno-cast-align -Wno-declaration-after-statement -fPIC -Wall -fvisibility=hidden -Wimplicit-function-declaration -Wredundant-decls -fno-omit-frame-pointer
Compiler:            GNU, 11.3.1
Target architecture: x64

**Comment by akallabeth** [comment]
@Teraya does that happen also if you skip `/mouse:relative:on,grab:on +mouse-relative` options?
I´d think so as this is expected for these options ;)
it you enable relative mouse updates the movement depends on acceleration settings which might be quite different between local and remote host.

**Comment by Teraya** [comment]
@akallabeth 
yeah, it's worse without the relative setting, (stays stuck in bottom right corner)
I haven't tried with a local machine, only with the AVD
Yes, it's a full remote desktop.  
I'll review the .rdp settings and reply back here again, but I don't recall anything too strange there that I noticed.  (I tried turning off "smart sizing", but still same issue).

**Comment by Teraya** [comment]
@akallabeth 
Minus the host/gw stuff, I have these...  any of these possibly related ?  

audiocapturemode:i:1
audiomode:i:0
authentication level:i:1
autoreconnection enabled:i:0
bandwidthautodetect:i:1
camerastoredirect:s:*
compression:i:1
devicestoredirect:s:
drivestoredirect:s:
enablecredsspsupport:i:1
encode redirected video capture:i:1
gatewaybrokeringtype:i:1
gatewaycredentialssource:i:0
gatewayprofileusagemethod:i:1
gatewayusagemethod:i:1
geo:s:US
networkautodetect:i:1
promptcredentialonce:i:1
redirectclipboard:i:1
redirectcomports:i:0
redirected video capture encoding quality:i:2
redirectprinters:i:0
redirectsmartcards:i:0
remoteapplicationmode:i:0
resourceprovider:s:arm
screen mode id:i:1
singlemoninwindowedmode:i:1
smart sizing:i:1
usbdevicestoredirect:s:*
use multimon:i:0
videoplaybackmode:i:1

**Comment by akallabeth** [comment]
@Teraya no, they are not related.

I´ve had a look at the mouse handling and there were a few issues with that.
https://github.com/FreeRDP/FreeRDP/pull/9609 should address them, can you test that?

**Comment by loligans** [comment]
@akallabeth This is the exact same issue I'm having. I have built and tested the latest release 3.0.0 (a34fa7c)

When I use the `xfreerdp`, the mouse is stuck on the corner, even with the relative mouse option turned on:
`./xfreerdp ~/Downloads/rdp/test.rdp /u:"redacted" /p:"redacted" /mouse:relative:on,grab:off +mouse-relative /w:1280 /h:720`

However when I switch to `sdl-freerdp` with the same exact CLI args, the cursor partially works, but instead becomes offset. Refer to the screenshot below. Where you see my cursor is exactly where I performed a right click and the context menu that appears is where it registers on the remote desktop.
![image](https://github.com/FreeRDP/FreeRDP/assets/3218210/fd3357bb-2481-45f7-9933-69caed651c49)

My build configuration:
```
cmake -GNinja \              
    -B freerdp-build \
    -S . \
    -DCMAKE_BUILD_TYPE=Release \
    -DCMAKE_SKIP_INSTALL_ALL_DEPENDENCY=ON \
    -DCMAKE_INSTALL_PREFIX=./out \
    -DWITH_SERVER=ON \
    -DWITH_SAMPLE=ON \
    -DWITH_PLATFORM_SERVER=OFF \
    -DUSE_UNWIND=OFF \
    -DWITH_SWSCALE=OFF \
    -DWITH_FFMPEG=OFF \
    -DWITH_WEBVIEW=OFF
```
